File: solvers.matlab/unsynch_relpose/test_num_unsynch_relpose.m
rng(23);
N = 10000;
nRoots = 16;
iterRef = 0;

stats = ini_stats('unsynch_relpose');
stats.solver = 'red_59x79_unsynch_relpose';

for i = 1:N

    tic;
    data = inidata_num_unsynch_relpose();
    [C,U,dU] = coefs_unsynch_relpose(data);

    S = feval(stats.solver,C);
    if isempty(S); continue; end
    [S,Ir,e] = validate_roots(S,C,U,dU,nRoots,iterRef);

    stats.tm = [stats.tm toc];
    stats = update_stats_o(stats,Ir,e);

end

disp_stats(stats,'plotHist',1,'saveHist',1);

File: solvers.julia/unsynch_relpose/test_num_unsynch_relpose.jl
include("../_common/header.jl")
include("inidata_num_unsynch_relpose.jl")
include("coefs_unsynch_relpose.jl")
include("red_59x79_unsynch_relpose.jl")

function test_num_unsynch_relpose()

    Random.seed!(23)
    N = 10000
    d = 16

    stats = ini_stats("unsynch_relpose")

    for i in 1:N
        data = inidata_num_unsynch_relpose()
        C,U = coefs_unsynch_relpose(data)

        toc = @elapsed S = red_59x79_unsynch_relpose(C)
        stats.tm = [stats.tm; toc]
        if isempty(S); continue; end

        stats = update_stats(stats,S,C,U,d)
    end

    disp_stats(stats)

end

test_num_unsynch_relpose()